qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/web/views/CreateAffiliationView.groovy

```diff
@@ -50,22 +51,43 @@ class CreateAffiliationView extends VerticalLayout {
     }
 
     private void initLayout() {
-        this.organisationField = new TextField("Organisation Name")
-        organisationField.setPlaceholder("name of the organisation")
-        organisationField.setDescription("Please enter the name of the organisation e.g. Universität Tübingen.")
-
-        this.addressAdditionField = new TextField("Department")
-        addressAdditionField.setPlaceholder("optional department name")
+        this.organisationBox = new ComboBox<>("Organisation Name")
+        
+        // we don't need the whole affiliation object, just the unique organization names.
+        List<String> organisationNames = sharedViewModel.affiliations.stream().map(affi -> affi.organisation).distinct().collect(Collectors.toList())
```

Comment:
Does the reload work here when the list of affiliations is modified?

Reply:
afaik this works because everything is cleared and loaded again upon registration of a new affiliation. let me quickly test this.

Reply:
This works, if that was the case you were thinking about!

Reply:
Ah yes, this might be the case since the affiliation resource services always throw away everything. But they still use the same list so I am not totally sure this works.

Reply:
tested it by creating a new organisation and was able to select that new organisation after the fields were cleared on registration. was able to create another one using that organisation (with other address addition).